app/scripts/lib/experiments/open-gmail.js

```diff
@@ -0,0 +1,62 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+define([
+  'underscore',
+  'lib/experiments/base'
+], function (_, BaseExperiment) {
+  'use strict';
+
+  var EXPERIMENT_NAME = 'openGmail';
+  var GMAIL_PATH = 'https://mail.google.com/mail/u/?authuser=';
+
+  function OpenGmailExperiment() {
+    // constructor
+  }
+
+  _.extend(OpenGmailExperiment.prototype, new BaseExperiment(), {
+    initialize: function (options) {
+      if (! options || ! options.account) {
+        return false;
+      }
+
+      options.extraAbleOptions = {
+        email: options.account.get('email')
+      };
+
+      return this.start(EXPERIMENT_NAME, options);
+    },
+    getContext: function () {
+      return {
+        isOpenGmailButtonVisible: this.isGroup('treatment')
+      };
+    },
+    performAction: function (action) {
+      if (action === 'openGmailTab') {
+        this.saveState('clicked');
+        this._openGmailTab();
+
+      } else if (action === 'triggered') {
+        this.saveState('triggered');
+
+      } else if (action === 'verification') {
+        this.saveState('verified');
+
+        if (this.hasState('clicked')) {
+          this.saveState('clicked.verified');
+        }
+
+        if (this.hasState('triggered')) {
+          this.saveState('triggered.verified');
+        }
+      }
+    },
+    _openGmailTab: function () {
+      var openWindow = this.window.open(GMAIL_PATH + this.email, '_blank');
```

Comment:
Adding gmail opening logic to an experiment file mixes experiment state logic with view logic and makes it more complex to convert an experiment to a full feature. How can these be teased apart so that new view related logic can live in a view related file (either a View itself, or a view mixin) so when an experiment has run its course and is no longer needed but the feature lives on, the feature code remains in place, but the experiment code is jettisoned? @TDA and I faced the same issue with the password strength checker. We placed the view related logic [into a mixin](https://github.com/mozilla/fxa-content-server/blob/master/app/scripts/views/mixins/password-strength-mixin.js). Maybe that concept can be refined further for this use case.